### PR TITLE
USHIFT-2261 USHIFT-2459: Add namespace ownership support for router

### PR DESCRIPTION
**Pull request by pacevedom**
**Which issue(s) this PR addresses**:

Closes #<Issue Number>

### CODE REVIEW

**Comment by pacevedom** [comment]
wait until https://github.com/openshift/microshift/pull/3066 is merged

**Comment by dhellmann** [comment]
It looks like this PR includes the contents of #3066, is that right? If so, that's good, I'd like to land the code change and tests at the same time. Maybe we can close the other PR?

**Comment by pacevedom** [comment]
> It looks like this PR includes the contents of #3066, is that right? If so, that's good, I'd like to land the code change and tests at the same time. Maybe we can close the other PR?

Yes, I was hoping for a quick review of the implementation (as its simple) and then land the tests later (which are more complex). To have them pass the tests I included all the other commits here too and planned to remove them once #3066 merged.

**Comment by ShudiLi** [comment]
The function worked well: By default, the ROUTER_DISABLE_NAMESPACE_OWNERSHIP_CHECK in the router deployment was "true", two route with same host names but with different path were admitted and could curl the routes successfully.
Changed it to "false", one route was  admitted, the other wasn't.

1.
% oc -n openshift-ingress get deployment -oyaml | grep  -A1 ROUTER_DISABLE_NAMESPACE_OWNERSHIP_CHECK 
          - name: ROUTER_DISABLE_NAMESPACE_OWNERSHIP_CHECK
            value: "true"
2.
% oc -n test2 get route
NAME           HOST                                  ADMITTED   SERVICE        TLS
route1         test-route1.apps.example.com          True       unsec-apach2   


3.
% oc -n test get route 
NAME           HOST                                 ADMITTED   SERVICE        TLS  
route1         test-route1.apps.example.com         True       unsec-apach2   
% oc -n test get route route1  -oyaml | grep -A2 spec:
spec:
  host: test-route1.apps.example.com
  path: /srv/docs

4.  curl the two routes
sh-4.4# curl http://test-route1.apps.example.com --resolve test-route1.apps.example.com:80:10.42.0.16
this a test!
sh-4.4# curl http://test-route1.apps.example.com/srv/docs/book.txt --resolve test-route1.apps.example.com:80:10.42.0.16
11111ss
sh-4.4#

5. also checked for edge route and reencrypt route

**Comment by ShudiLi** [comment]
/label qe-approved
thanks

File: cmd/generate-config/config/config-openapi-spec.json [hunk 1]
```diff
@@ -69,6 +70,26 @@
         }
       }
     },
+    "ingress": {
```
Comment (jerpeter1) [inline cmd/generate-config/config/config-openapi-spec.json:73]: In your EP, you said you intended to expose this via:

```yaml
router:
  routerAdmissionPolicy:
    namespaceOwnership: <Strict|InterNamespaceAllowed> # Defaults to InterNamespaceAllowed.
```

here, you're exposing it via ingress.routerAdmissionPolicy.namespaceOwnership - if the implementation is what we want, you should probably go back and fixup the EP.

**Comment by jerpeter1** [comment]
You might consider adding a test case which checks that the default (unconfigured) code path works as expected. Maybe another one for invalid configuration, but I wouldn't consider either test a blocker for merging this.

/lgtm

File: cmd/generate-config/config/config-openapi-spec.json (hunk 1 above)
Comment (pacevedom) [reply cmd/generate-config/config/config-openapi-spec.json:73]: Good catch, will update it right away.

**Comment by pacevedom** [comment]
> You might consider adding a test case which checks that the default (unconfigured) code path works as expected. Maybe another one for invalid configuration, but I wouldn't consider either test a blocker for merging this.
> 
> /lgtm

Will add them in a separate PR. Thanks!

**Pull request merged**